nep-11.mediawiki

```diff
@@ -0,0 +1,416 @@
+If so, the transfer SHOULD be processed; If not, the function SHOULD use the SYSCALL <code>Neo.Runtime.CheckWitness</code> to verify the transfer.
+
+If the transfer is not processed, the function SHOULD return <code>false</code>.
+
+====ownerOf====
+
+<pre>
+{
+  "name": "ownerOf",
+  "parameters": [
+    {
+      "name": "tokenid",
+      "type": "ByteString"
+    }
+  ],
+  "returntype": "InteropInterface<iterator>"
+}
+</pre>
+
+Returns an <code>iterator</code> that contains all the co-owners that own the specified token.
+
+The parameter <code>tokenid</code> SHOULD be a valid NFT.
+If not, this method SHOULD <code>throw</code> an exception.
+
+====balanceOf====
+
+<pre>
+{
+  "name": "balanceOf",
+  "parameters": [
+    {
+      "name": "owner",
+      "type": "Hash160"
+    },
+    {
+      "name": "tokenid",
+      "type": "ByteString"
+    }
+  ],
+  "returntype": "Integer"
+}
+</pre>
+
+This method returns the balance of the specified token for the specified <code>owner</code>'s account.
+
+The parameter <code>tokenid</code> SHOULD be a valid NFT.
+If not, this method SHOULD <code>throw</code> an exception.
+
+The parameter <code>owner</code> SHOULD be a 20-byte address.
+If not, this method SHOULD <code>throw</code> an exception.
+
+If the <code>owner</code> is an unused address, or it's not the owner of the specified token, this method SHOULD return <code>0</code>.
+
+===Optional methods===
+
+====tokens====
+
+<pre>
+{
+  "name": "tokens",
+  "parameters": [],
+  "returntype": "InteropInterface<iterator>"
+}
+</pre>
+
+Returns an <code>iterator</code> that contains all of the tokens minted by the contract.
+
+====properties====
+
+<pre>
+{
+  "name": "properties",
+  "parameters": [
+    {
+      "name": "tokenid",
+      "type": "ByteString"
+    }
+  ],
+  "returntype": "Array"
+}
+</pre>
+
+Returns a serialized NVM object containing the properties for the given NFT.
+The NVM object must conform to the "NEO NFT Metadata JSON Schema".
+
+The parameter <code>tokenid</code> SHOULD be a valid NFT.
+If not, this method SHOULD <code>throw</code> an exception.
+
+Following is a description of the JSON Schema:
+<pre>
+{
+  "title": "Asset Metadata",
+  "type": "object",
+  "properties": {
+    "name": {
+      "type": "string",
+      "description": "Identifies the asset to which this NFT represents."
+    },
+    "description": {
+      "type": "string",
+      "description": "Describes the asset to which this NFT represents."
+    },
+    "image": {
+      "type": "string",
+      "description": "Optional. A URI pointing to a resource with mime type image/* representing the asset to which this NFT represents. Consider making any images at a width between 320 and 1080 pixels and aspect ratio between 1.91:1 and 4:5 inclusive."
+    },
+    "tokenURI": {
+      "type": "string",
+      "description": "Optional. A distinct URI for a given asset that adheres to RFC 3986"
```

Comment:
Can be `"format": "uri"` (1) if corresponding vocabulary is included (2).
https://json-schema.org/draft/2019-09/json-schema-validation.html#rfc.section.7.3.5
https://github.com/json-schema-org/json-schema-spec/blob/master/schema.json#L10

Reply:
Done.